fdbserver/TenantBalancer.actor.cpp

```diff
@@ -1689,22 +1743,69 @@ ACTOR Future<Void> abortMovement(TenantBalancer* self, AbortMovementRequest req)
 	    .detail("Prefix", req.prefix);
 
 	try {
-		state Reference<const MovementRecord> record =
-		    self->getMovementSnapshot(req.movementLocation, req.prefix, req.movementId);
+		state Reference<MovementRecord> record = self->getMovementSnapshot(req.movementLocation, req.prefix)->clone();
```

Comment:
I don't think we need to clone the movement record:

```suggestion
		state Reference<MovementRecord> record = self->getMovementSnapshot(req.movementLocation, req.prefix);
```

If the problem is that we want to get rid of the `const` for later usages, then we could either name this something else or even make it anonymous:

self->getMovementSnapshot(req.movementLocation, req.prefix)->abort();

Reply:
In this abort implementation function, we need to do the followings:
```
record->movementState = MovementState::COMPLETED;
wait(self->saveMovementRecord(mutableRecord));

```
Although we can use const_cast to remove the const there, but I'm not sure if it will cause some bugs to other in-flight operations

Reply:
My idea here is to have two different variables for the record, one for the const record and another for the mutable copy. The first could even be anonymous, we don't need to give it a name.

Reply:
I guess I know how to solve it